src/main.rs

```diff
@@ -195,6 +202,15 @@ where
                                 display_handler.unpark();
                                 break;
                             }
+                            Event::Key(Key::Char(' ')) => {
+                                // Xor with true is the same as negation
+                                // fetch_xor() returns old value, so negate it
+                                // to get the updated pause variable
+                                let new_paused = !paused.fetch_xor(true, Ordering::SeqCst);
+                                if let Ok(mut ui) = ui.lock() {
```

Comment:
I think here we can just `display_handler.unpark();` to wake up the display thread. This way we avoid modifying the UI state from the input handling thread.

Reply:
Changed